### ISSUE TITLE
Make configuration available from `utils`

### ISSUE THREAD

**Issue by wmertens** [xaviergonz/makfy#2]
I'd like to autogenerate my `package.json` from `makfy`, including adding the `makfy` scripts into it.

To do that, I would need to read the current `makfyfile`, but I don't necessarily know where that it, so it would be nice if there was a `getCommands()` function which simply returns the current configuration…

**Comment by wmertens** [comment]
Also related: knowing if makfy is in `color` mode or not. That way, it can be passed to e.g. `webpack`.

**Comment by xaviergonz** [comment]
@wmertens thanks for your interest :) however I'm not sure I follow, what would you want the output of that method to be like? (I don't quite get your aim exactly)

as for color you can use https://www.npmjs.com/package/chalk chalk.supportsColor, since that's what makfy internally uses to colorize text

**Comment by xaviergonz** [comment]
I think I get it now, couldn't you do something like:

const commands = { ... }

and on one of the commands (say buildPackageJson) something like

```js
// load package.json and parse into packageJsonObj
packageJsonObj.scripts = {};
Object.keys(commands).forEach((cmdName) => {
  packageJsonObj[cmdName] = `makfy ${cmdName}`;
});
// stringify and save back
```

**Comment by wmertens** [comment]
Yes that's what I want to do, but I don't know what the commands are, I
don't control the makfyfile from this lib. So I'd like that utils.commands
or utils.getCommands() is the normalized command object.

**Comment by wmertens** [comment]
And as for the color, makfy is already running `chalk` to check the color, as well as accepting command line parameters, so the only one that really knows if colors should be used is makfy.

It would be nice if that is available, maybe under `utils.makfyOptions.color`?

**Comment by xaviergonz** [comment]
What do you mean by "I don't control the makfyfile from this lib."? what should utils.getCommands() do, return the keys of the command object? return the commands object itself?

Actually chalk.supportsColor is based on what's passed on the command line as well (chalk is the one that parses the command line for a --color or --no-color argument + if there's no command line argument it tries to autodetect if the terminal supports it and sets that supportsColor accordingly). However I see no harm in exposing it... 

would utils.makfyExports.commands + utils.makfyExports.options suit your needs?

**Comment by xaviergonz** [comment]
Ok, I just released 1.1.11, check makfyContext on the docs.
Does that do what you want?

**Comment by wmertens** [comment]
Yes that is exactly what I needed 👯 

No links yet, but soonish. I'm working on a framework for general applications (focused on React+Redux+GraphQL+SSR+hot reload server/client, but each of those parts could potentially be replaced) that uses plugins to be very configurable.

It's quite hard to come up with the right abstractions, but your project is definitely one of the right abstractions :)
